core/src/mindustry/world/blocks/defense/ShockMine.java

```diff
@@ -37,20 +40,30 @@ public void drawTeam(){
         @Override
         public void draw(){
             super.draw();
-            Draw.color(team.color);
-            Draw.alpha(0.22f);
-            Fill.rect(x, y, 2f, 2f);
-            Draw.color();
+            if(isActive() || progress < 20f){
+                Draw.color(team.color);
+                Draw.alpha(0.5f);
+                Fill.rect(x, y, 2f, 2f);
+                Draw.color();
+            }
+            if(!isActive()){
+                progress += delta();
+                progress %= blinkDelay;
+            }
```

Comment:
it's better to use `else` here.
```suggestion
            }else{
                progress += delta();
                progress %= blinkDelay;
            }
```

Reply:
this will break stuff